src/connection.cpp

```diff
@@ -491,6 +510,7 @@ void MatrixConnection::onSyncDone()
     }
 
     m_contactListIface->setContactListState(Tp::ContactListStateSuccess);
+    m_connection->sync();
```

Comment:
Why this is needed? It looks wrong (and maybe traffic-expensive) to sync again on sync done.
Do we need to sync if we're online? Maybe the thing is that it is in the Matrix (or QMatrixClient) nature to lose the connection during a session and connect again and again. I hope that in this case, we can request sync on the right event — "on connection established" or something like that.

Reply:
contributed to libqmatrixclient https://github.com/QMatrixClient/libqmatrixclient/pull/274
version 0.5 of the library have `syncLoop` method